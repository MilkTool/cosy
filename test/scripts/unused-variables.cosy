// var a = 5 // not reported unused because its top-level
var b = 6
print b

{
    var c = 5 // c: unused variable
    var d = 6
    print d
}

fn x(arg1) {
    var local1 // local1: unused variable
    print arg1
}

fn y(arg2) { // arg2: unused variable
    var local2
    print local2
}

x(true)
y(false)

// ----------

fn f(_x) { // no error for unused argument
    var _y // no error for unused variable
    var _z = 5
    print _z // error for used variable starting with _
}

f('hello')
