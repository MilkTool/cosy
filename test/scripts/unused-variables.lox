var a = 5 // not reported unused because its top-level
var b = 6
print b

{
    var c = 5 // c: unused variable
    var d = 6
    print d
}

fun x(arg1) {
    var local1 // local1: unused variable
    print arg1
}

fun y(arg2) { // arg2: unused variable
    var local2
    print local2
}
