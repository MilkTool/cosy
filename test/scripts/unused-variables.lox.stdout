[line 6] Error at "c": Local variable is not used.
[line 12] Error at "local1": Local variable is not used.
[line 16] Error at "arg2": Local variable is not used.
